### DISCUSSION TITLE
getSession (v2) and local storage

### DISCUSSION THREAD

**Discussion by Nemmcy** [supabase/supabase#8595] [Questions] [answered]
I don't understand what updates the session stored in local storage, i.e. what function triggers the update of the access token using the refresh token.

The session object in local storage is updated anytime we use one of the signup/signin/signout functions, but what else?

- Does using supabase.auth.getSession() automatically checks if the access token is valid, performs a token refresh if it isn't and updates the session object in local storage?

- Is the session object in local storage automatically updated every time a component using any supabase function re-renders?

**Comment by GaryAustin1** [comment, marked as answer]
getSession does refresh the token.
There is also a timer that runs to refresh the access token in the background shortly before it expires (unless you turn it off in client settings).
There is no refresh when normal functions are called.

You can watch .onAuthStateChange() to see the events... https://supabase.com/docs/reference/javascript/next/auth-onauthstatechange
